packages/@aws-cdk/core/lib/runtime.ts

```diff
@@ -217,7 +217,7 @@ export function validateString(x: any): ValidationResult {
 }
 
 export function validateNumber(x: any): ValidationResult {
-  if (canInspect(x) && typeof x !== 'number') {
+  if (canInspect(x) && (typeof x !== 'number' && typeof x !== 'string')) {
```

Comment:
This is definitely not enough.

A string that doesn't parse to a decimal number (like `"abc") should not be correctly validated here.